assets/javascripts/discourse/initializers/patreon.js.es6

```diff
@@ -0,0 +1,56 @@
+import { withPluginApi } from "discourse/lib/plugin-api";
+import TopicRoute from "discourse/routes/topic";
+
+let numTopicsOpened = 0;
+
+function initWithApi(api) {
+  const currentUser = api.getCurrentUser();
+
+  TopicRoute.reopen({
+    setupController(controller, model) {
+      this._super(controller, model);
+
+      if (!currentUser) return;
+
+      Ember.run.scheduleOnce("afterRender", () => {
+        // only count regular topics
+        if (model.get("isPrivateMessage")) return;
```

Comment:
Shouldn't that test be outside the `scheduleOnce` call?

Reply:
Based on my observation, the `model` object is not always available immediately but always avalable on `afterRender`. Perhaps I'm wrong?